ansible-galaxy - support `resolvelib >= 0.5.3, < 1.1.0`

hswong3i opened the pull request:
##### SUMMARY
<https://pypi.org/project/resolvelib/1.0.1> released on 2023-03-09:

-   <https://github.com/sarugaku/resolvelib/blob/main/CHANGELOG.rst#101-2023-03-09>
-   <https://github.com/sarugaku/resolvelib/releases/tag/1.0.1>
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
API guarantees discussions:
* https://github.com/sarugaku/resolvelib/issues/97
* https://github.com/sarugaku/resolvelib/pull/125

hswong3i committed:
ansible-galaxy - support `resolvelib >= 0.5.3, < 1.1.0`

<https://pypi.org/project/resolvelib/1.0.1> released on 2023-03-09:

-   <https://github.com/sarugaku/resolvelib/blob/main/CHANGELOG.rst#101-2023-03-09>
-   <https://github.com/sarugaku/resolvelib/releases/tag/1.0.1>

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>

mattclay commented on test/lib/ansible_test/_data/requirements/ansible.txt:
```diff
@@ -15,4 +15,4 @@ importlib_resources >= 5.0, < 5.1; python_version < '3.10'
 # NOTE: When updating the upper bound, also update the latest version used
 # NOTE: in the ansible-galaxy-collection test suite.
-resolvelib >= 0.5.3, < 0.10.0  # dependency resolver used by ansible-galaxy
+resolvelib >= 0.5.3, < 1.1.0  # dependency resolver used by ansible-galaxy
```
To verify the new version works, temporarily change this to `resolvelib==1.0.1`.

mattclay commented on requirements.txt:
```diff
@@ -15,4 +15,4 @@ importlib_resources >= 5.0, < 5.1; python_version < '3.10'
 # NOTE: When updating the upper bound, also update the latest version used
 # NOTE: in the ansible-galaxy-collection test suite.
-resolvelib >= 0.5.3, < 0.10.0  # dependency resolver used by ansible-galaxy
+resolvelib >= 0.5.3, < 1.1.0  # dependency resolver used by ansible-galaxy
```
To verify the new version works, temporarily change this to `resolvelib==1.0.1`.

webknjaz replied:
I'd rather extend https://github.com/ansible/ansible/blob/bad8843/test/integration/targets/ansible-galaxy-collection/vars/main.yml#L7-L11 with more versions.

mattclay replied:
That should be updated as well.

@webknjaz Do you know if that covers all of our uses of resolvelib? If not, the temporary commit and revert would still be useful.

hswong3i replied:
also update for https://github.com/ansible/ansible/blob/bad8843/test/integration/targets/ansible-galaxy-collection/vars/main.yml#L7-L11

webknjaz replied:
@mattclay I don't know. It's best to ask @s-hertel.

mattclay replied:
@hswong3i Tests passed with the pinned version. Go ahead and revert that commit. No need to force push the revert, we'll just squash the commits when merging the PR.

hswong3i replied:
done ;-)

mattclay reviewed (changes requested):
Temporary changes are needed to verify the new version passes in CI. Once it passes, those changes can be reverted.

hswong3i committed:
Trigger CI by pinning resolvelib with latest version

Since resolvelib is pre-installed in our test containers, we should temporarily pin the latest version allowed to force the tests to run with that version. Once the tests have passed that commit can be reverted.

Please make those changes without force pushing, so that we keep the reference to the passing CI run. We can squash the commits when merging the PR so the temporary commits won't be in the final commit merged to the devel branch.

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>

hswong3i committed:
https://github.com/ansible/ansible/pull/80196#discussion_r1136003637

Also test resolvelib with multiple supported versions.

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>

hswong3i committed:
Revert "Trigger CI by pinning resolvelib with latest version"

This reverts commit 5518e5dbca01b87d13850cee9ccdbdc1ee49188f.
